core/src/changes-vtab-read.c

```diff
@@ -76,7 +76,7 @@ char *crsql_changesUnionQuery(crsql_TableInfo **tableInfos, int tableInfosLen,
       "SELECT tbl, pks, cid, col_vrsn, db_vrsn, site_id, opid FROM (%z) "
       "%s%s ORDER "
       "BY "
-      "db_vrsn, tbl ASC",
+      "opid ASC",
```

Comment:
this (along with all the prior changes required to bring about opid) should solve the `within a transaction` ordering problem once it is well tested and merged 

cc @Azarattum

Reply:
This technically doesn't add any additional space to the clock tables since it replaces sqlite's implicit `rowid` column.

Reply:
You could also use this value to paginate sync within a transaction if you have a very large transaction that shouldn't be sent in a single message.

Reply:
See an example of rowid from crsql_changes here:

https://github.com/vlcn-io/cr-sqlite/pull/221/commits/a8c4c98d27c4a47be34c0b576536636fb4f1ea80#diff-e1f437622b39279fa16a81899fb6f0600afd0c66e71e62d71666817d65dd6658R68-R101

Reply:
Looks great! Do we have to `ORDER BY opid` or `ORDER BY rowid` when doing a `SELECT FROM crsql_changes` or since it's the rowid it would be a default ordering?

Reply:
it'll be sorted by rowid by default.